src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -339,18 +361,26 @@ internal bool TryWrite(Utf8JsonWriter writer, in T value, JsonSerializerOptions
                     JsonConverter jsonConverter = state.Current.InitializeReEntry(type, options);
                     if (jsonConverter != this)
```

Comment:
Is that possible? I was trying to think in a case where this could be true but couldn't think of any. 

We hit this condition only if the JsonConverter is our internal converter for `object` (`ObjectConverter`). 

When we use `InitializeReEntry` to get the converter for the runtime type, is it possible that it returns the  ObjectConverter? I would imagine that it would return a converter for a type other than `object` given that we previously verified that `type != typeof(object)`.

https://github.com/dotnet/runtime/blob/1905462b9b48b694eb898941d4f1019c14dbdce5/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs#L350-L355
I suspect that this condition used to make sense before https://github.com/dotnet/runtime/pull/40914 but now is pointless.

Reply:
Makes sense to change this to `Debug.Assert(jsonConverter != this);` in a follow up.